### PR TITLE
Turn Form Search into Supervisor Portal

### PR DESCRIPTION
**Pull request by bledsoef**
Fixes issue #362 
## Changes
- **Removed** the existing **Supervisor Portal** along with **My Students** and **Department Students**.
- Changed the title of **Form Search to Supervisor Portal**.
- **Made** the **form search** bit **an accordion** that contracts when a search is processed so it only displays the search result table.
- **Removed** all instances of **index.js, index.html, and index.css**.
- Replaced formSearch.py, formSearch.js, and formSearch.html with supervisorPortal.py, etc.
- Fixed a variety of bugs in the form search mechanism.
- When the **Actions** dropdown only displays **Labor History**, the **Actions** dropdown is replaced with a button that says **Labor History** and is a direct link to the **Labor History Modal**.
- Made the student's name link to studentHistoryModal.
- Combined Form Type and Form Status into one column
- Got rid of the actions button in favor of a direct link.
- Added Department ID (ORG + ACCOUNT) to the studentHistoryModal.

## To Test
- `./reset_database from-backup`, `export FLASK_ENV=staging`, and `flask run`
- Open LSF in a window.
- Ensure that the existing Supervisor Portal tab leads you to the old Form Search page instead and that it no longer leads you to the old Supervisor Portal.
- Execute a search without selecting any options and make sure you get an error and make sure the accordion collapses.
- Execute another search with any filters selected and ensure that you can access a student's labor history modal if their form was approved, or have the ability to select an action otherwise.
- `./tests/run_tests.sh`

### CODE REVIEW

**Comment by sreynit02** [comment]
When the accordion is clicked on, the corner of the accordion looks a bit crooked:
image

**Review by AndersonStettner — CHANGES_REQUESTED** [review]
The new page is looking good. A few things:
1. Make sure the accordion stays open when the user searches with none of the fields filled in and the "At least one field..." error message is thrown.

2. In the "Department" dropdown the second option is a blank space but if you select it, it says "Department". If it is not supposed to be there remove it and if it is supposed to be there make sure "Department" shows up in the drop down.
Figures:  
image
image

3. Add an indication that the Form Table is separate from the "Form Search" accordion. 

4. Is the arrow on the right of the accordion body supposed to indicate open state of the accordion? If so then it is broken, if not then it might be nice to remove it since it could be confusing to users.

5. What is the purpose of the green indicator to the left of "Labor Status Form" in a students LSF modal? In my opinion the does not look very good due to how pixelated it is and there does not seem to be a purpose for it since there is no differentiation when a LSF is submitted, accepted, or rejected. 
Figure:
image

File: app/logic/search.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+from app.models.laborStatusForm import LaborStatusForm
+from app.models.department import Department
+from app.models.formHistory import FormHistory
+
+def limitSearchByUserDepartment(students, currentUser):
+    """
+    Given a list of student dictionaries and the currentUser the function will only return a
+    list of students who have held a labor position within the currentUsers department.
+    """
+    newstudents = []
```
Comment (AndersonStettner) [inline app/logic/search.py]: camelCase?

File: app/logic/search.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    """
+    departments = (Department.select()
+                    .join_from(Department, LaborStatusForm)
+                    .join_from(LaborStatusForm, FormHistory)
+                    .join_from(LaborStatusForm, Supervisor)
+                    .where((LaborStatusForm.supervisor.ID == currentUser.supervisor.ID) | (FormHistory.createdBy == currentUser)).order_by(Department.DEPT_NAME)
+                    .distinct()
+                    )
+
+    return departments
```
Comment (AndersonStettner) [inline app/logic/search.py]: Make the return of this function and the previous function (studentDbToDict) consistent. They should both either be variables or just return the query for this function.

**Review by AndersonStettner — CHANGES_REQUESTED** [review]

**Comment by tylerpar99** [comment]
> The new page is looking good. A few things:
> 
> 1. Make sure the accordion stays open when the user searches with none of the fields filled in and the "At least one field..." error message is thrown.
> 2. In the "Department" dropdown the second option is a blank space but if you select it, it says "Department". If it is not supposed to be there remove it and if it is supposed to be there make sure "Department" shows up in the drop down.
>    Figures:
>    image
>    image
> 3. Add an indication that the Form Table is separate from the "Form Search" accordion.
> 4. Is the arrow on the right of the accordion body supposed to indicate open state of the accordion? If so then it is broken, if not then it might be nice to remove it since it could be confusing to users.
> 5. What is the purpose of the green indicator to the left of "Labor Status Form" in a students LSF modal? In my opinion the does not look very good due to how pixelated it is and there does not seem to be a purpose for it since there is no differentiation when a LSF is submitted, accepted, or rejected.
>    Figure:
>    image

1. Done.
2. DB issue is not a code issue, there is a department record in the backup DB that has an empty name.
3. Done.
4. Done.
5. Done.

**Comment by tylerpar99** [comment]
> When the accordion is clicked on, the corner of the accordion looks a bit crooked: image

fixed.

**Review by AndersonStettner — CHANGES_REQUESTED** [review]
It looks good, one minor thing I noticed is that, if a user does not select a field and they get the "At least one field must be selected." flash message, the flash message never goes away unless the user closes it themselves. 

Also, a nitpick I have is I don't really like how the accordion behaves in that case, searching with no fields selected. I think it might be good to disable the search button until one of the fields has been selected or have a different action happen when that case occurs.

**Comment by tylerpar99** [comment]
> It looks good, one minor thing I noticed is that, if a user does not select a field and they get the "At least one field must be selected." flash message, the flash message never goes away unless the user closes it themselves.
> 
> Also, a nitpick I have is I don't really like how the accordion behaves in that case, searching with no fields selected. I think it might be good to disable the search button until one of the fields has been selected or have a different action happen when that case occurs.

Fixed.

**Review by martinj2-dot — APPROVED** [review]
Looks good!

**Review by Karina-Agliullova — APPROVED** [review]
Good work!

File: app/logic/search.py [hunk 3]
```diff
@@ -34,11 +34,12 @@ def studentDbToDict(student):
     """
     Given a student object it will return a mapped Dict with student data.
     """
-    return {'username': usernameFromEmail(student.STU_EMAIL.strip()),
-            'firstName': student.FIRST_NAME.strip(),
-            'lastName': student.LAST_NAME.strip(),
-            'bnumber': student.ID.strip(),
-            'type': 'Student'}
+    dbtodict =  {'username': usernameFromEmail(student.STU_EMAIL.strip()),
```
Comment (AndersonStettner) [inline app/logic/search.py]: camelCase?

Comment (tylerpar99) [reply app/logic/search.py]: Fixed.

**Review by AndersonStettner — APPROVED** [review]

**Review by BrianRamsay — APPROVED** [review]

**Pull request merged**